### ISSUE TITLE
v2.12.1 still does not build with Visual Studio (2015/2017)

### ISSUE THREAD

**Issue by janusw** [LLNL/hypre#19]
Trying to compile the recent version 2.12.1 (8724fc3d974963d8feb161aac03d2ad666e1b747) with Visual Studio, I still see several issues (that occur with both VS2015 and VS2017):

1. The problems reported in #14 persist (since 2.11.1).
2. The problems reported in #16 persist (since 2.12.0).
3. There is even one new issue that has not been seen with previous versions, namely that M_PI is undeclared here:

```
hypre\src\parcsr_ls\par_vardifconv_rs.c(501): error C2065: "M_PI": nichtdeklarierter Bezeichner
hypre\src\parcsr_ls\par_vardifconv_rs.c(519): error C2065: "M_PI": nichtdeklarierter Bezeichner
```

I hope these issues can be fixed for the upcoming version 2.13.0, so that there finally is an 'official' HYPRE version that can be built with Visual Studio.

Cheers,
Janus

**Comment by janusw** [comment]
@rfalgout

> There is even one new issue that has not been seen with previous versions, namely that M_PI is undeclared here

I fixed this in 66a7f0c850158647146335124a285959a33956fb.

My branch https://github.com/janusw/hypre/tree/v2.12.1-vs contains fixes for all three issues mentioned above. Please merge before making a new release (or fix these problems in your preferred manner).

Thanks,
Janus

**Comment by rfalgout** [comment]
@janusw This is not the release to look at.  Look for 2.13.0 when it appears.  Don't waste your time on this one; we know it doesn't work.  Thanks!

**Comment by janusw** [comment]
Hi Rob,

just tried 2.13.0 and indeed it appears to compile flawlessly with VS15. Thanks for finally fixing that stuff!

Cheers,
Janus

**Comment by rfalgout** [comment]
VS 2015 issues were resolved in version 2.13.0
